graphix/_db.py

```diff
@@ -2,43 +2,51 @@
 
 from __future__ import annotations
 
+from typing import ClassVar, TypeVar
+
 import numpy as np
 import numpy.typing as npt
 
+_T = TypeVar("_T", bound=np.generic)
+
 
-def _lock(m: npt.NDArray[np.complex128]) -> npt.NDArray[np.complex128]:
-    """Create a true immutable view."""
+def _lock(data: npt.NDArray[_T]) -> npt.NDArray[np.complex128]:
+    """Create a true immutable view.
+
+    data must be a "temporary" array, i.e., should not be bound to a variable.
+    """
+    m = data.astype(np.complex128)
     m.flags.writeable = False
```

Comment:
I'm using this strange design here for maximizing safety.
I admit your design is more concise, but it's also potentially unsafe.
If conciseness beats safety, `_lock` itself would be redundant.

Reply:
If this doesn't slow us down (I presume), it is perhaps ok as is, perhaps could you add short comments to the code what this is trying to avoid (no need to be detailed) so that people can understand intention later.

Reply:
Resolved in e48e7ecaa26cb747c5d64b0aebd76f60d3a6c6c4 .